### PR TITLE
Profile page added + new UI features

### PR DESCRIPTION
**Pull request by Bahaaib**
Track issue here:
https://github.com/ammaratef45/Attendance/issues/21

### CODE REVIEW

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#23](https://github.com/ammaratef45/attendance/issues/23) is now in scope, role is `REV`

File: attendance/lib/ProfilePage/profile_page.dart [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import 'package:flutter/material.dart';
+
+import './profile_page_view.dart';
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page.dart]: @Bahaaib please avoid relative imports, use package style instead

File: attendance/lib/ProfilePage/profile_page_view.dart [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import 'package:flutter/material.dart';
+
+import './profile_page_viewmodel.dart';
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib The relative import thing also

File: attendance/lib/main.dart [hunk 3]
```diff
@@ -1,12 +1,16 @@
 import 'package:flutter/material.dart';
-import './loginPage/login_page.dart';
-import './homePage/home_page.dart';
+
 import './AttendanceDatailsPage/attendance_details_page.dart';
-import './offline_page/offline_page.dart';
+import './ProfilePage/profile_page.dart';
+import './homePage/home_page.dart';
+import './loginPage/login_page.dart';
+import './offline_page/offlinine_page.dart';
```
Comment (ammaratef45) [inline attendance/lib/main.dart]: @Bahaaib this line is wrong, the name of the file is wrong

File: attendance/lib/ProfilePage/profile_page_view.dart [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+class ProfilePageView extends ProfilePageViewModel {
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib can you add an app bar? currently no up button

File: attendance/lib/ProfilePage/profile_page_view.dart [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  Container(
+                    width: 250,
+                    margin: EdgeInsets.only(top: 10.0),
+                    child: Theme(
+                      data: ThemeData(hintColor: Colors.lightBlueAccent),
+                      child: TextField(
+                        controller: controller,
+                        onChanged: (text) => print("First text field: $text"),
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib remove print and add a puzzle to update the user's native name

**Comment by ammaratef45** [comment]
@Bahaaib You've done pretty work, I like the design and the fixes you made
Please either implement my changes or leave puzzles for them
[More about PDD here](https://github.com/yegor256/0pdd).
You can check the puzzles will not complain by running `pdd -f /dev/null -v` locally.
submit a ticket if you had an issue with this.

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) everybody who has role `REV` is banned at [#23](https://github.com/ammaratef45/attendance/issues/23); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

File: attendance/lib/ProfilePage/profile_page_view.dart (hunk 4 above)
Comment (Bahaaib) [reply attendance/lib/ProfilePage/profile_page_view.dart]: The App bar here will blow up the page design

Comment (ammaratef45) [reply attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib the up button is necessary for all pages (except login and main page), that's a UX rule (we have iOS users who will kill us)

**Comment by ammaratef45** [comment]
@Bahaaib can you add the missed puzzles?
1- a todo to send entered data to the API.
2- a todo to include the native name in the attendance excel sheet.

**Comment by ammaratef45** [comment]
@rultor merge

File: attendance/lib/ProfilePage/profile_page_view.dart [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+                children: <Widget>[
+                  Container(
+                    width: 250,
+                    margin: EdgeInsets.only(top: 10.0),
+                    child: Theme(
+                      data: ThemeData(hintColor: Colors.lightBlueAccent),
+                      child: TextField(
+                        controller: controller,
+                        onChanged: (text) {
+                          //TODO: Update..
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib This todo is not a puzzle format

**Comment by ammaratef45** [comment]
@rultor merge

File: attendance/lib/ProfilePage/profile_page_view.dart (hunk 6 above)
Comment (Bahaaib) [reply attendance/lib/ProfilePage/profile_page_view.dart]: Oops. I added it days ago before knowing about pdd.

**Comment by 0crat** [comment]
Job `gh:ammaratef45/attendance#23` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#23](https://github.com/ammaratef45/attendance/issues/23) is now out of scope

**Pull request merged**